src/vs/workbench/common/editor/editorGroup.ts

```diff
@@ -663,6 +664,14 @@ export class EditorGroup extends Disposable {
 					&& !this.configurationService.getValue<boolean>('sql.promptToSaveGeneratedFiles')) {
 					return;
 				}
+				// Do not add generated files from Temp if file is not dirty
+				if (e instanceof FileEditorInput && !e.isDirty()) {
+					let filePath = e.getResource() ? e.getResource().fsPath : undefined;
```

Comment:
So the reason we're doing this is because the "Script as CREATE"/Go to Def/etc. are creating an actual file on disk? Why aren't we just using an UntitledEditorInput for that and have it just be a memory buffer, that seems to make more sense than writing actual files

Reply:
Script as create does not - it creats untitled. 
Only go to definition does because we override the vscode's go to definition which expects the file to be actually present (since that is true for most code definitions)